src/sema/tests/mod.rs

```diff
@@ -128,3 +139,193 @@ fn test_statement_reachable() {
+    
+        function test_types32(bytes input ) public {
+            // value 2147483648 does not fit into type int32.
+            int32 add_ovf = 2147483647 + 1;
+
+            // value 2147483648 does not fit into type int32.
+            int32 add_normal = 2147483647 + 0;
+
+            // value 2147483648 does not fit into type int32.
+            int32 mixed = 2147483647 + 1 + input.readInt32LE(2);
+        }
+    
+
+        function test_types64(bytes input ) public {
+            // value 9223372036854775808 does not fit into type int64.
+            int64 add_ovf = 9223372036854775807 + 1 ;
+
+            int64 add_normal = 9223372036854775807;
+
+            // value 9223372036854775808 does not fit into type int64.
+            int64 mixed = 9223372036854775807 + 1 + input.readInt64LE(2);
+
+            // value 18446744073709551616 does not fit into type uint64.
+            uint64 pow_ovf = 2 ** 64  ;
+
+            uint64 normal_pow = (2**64) -1 ;
+        }
+
+
+
+        function test_types_128_256(bytes input ) public {
+            
+            while (true) {
+                // value 340282366920938463463374607431768211456 does not fit into type uint64.
+                uint128 ovf = 2 ** 128 ;
+                uint128 normal = 2** 128 -1 ;
+            }
+            uint128[] arr;
+            // negative value -1 does not fit into type uint32. Cannot implicitly convert signed literal to unsigned type.
+            // value 340282366920938463463374607431768211456 does not fit into type uint64.
+            uint128 access = arr[1-2] + 1 + (2**128);
+            // value 340282366920938463463374607431768211456 does not fit into type uint128.
+            uint256 num = 3e255;
+            // exponent '256' too large
+            uint256 num = 3e256;
+        }
+        
+    }
+        "#;
+    let ns = parse(file);
+    let errors = ns.diagnostics.errors();
+
+    assert_eq!(
+        errors[0].message,
+        "value 2147483648 does not fit into type int32."
+    );
+    assert_eq!(
+        errors[1].message,
+        "value 2147483648 does not fit into type int32."
+    );
+    assert_eq!(
+        errors[2].message,
+        "value 9223372036854775808 does not fit into type int64."
+    );
+    assert_eq!(
+        errors[3].message,
+        "value 9223372036854775808 does not fit into type int64."
+    );
+    assert_eq!(
+        errors[4].message,
+        "value 18446744073709551616 does not fit into type uint64."
+    );
+    assert_eq!(
+        errors[5].message,
+        "value 340282366920938463463374607431768211456 does not fit into type uint128."
+    );
+    assert_eq!(errors[6].message, "negative value -1 does not fit into type uint32. Cannot implicitly convert signed literal to unsigned type.");
+    assert_eq!(
+        errors[7].message,
+        "value 340282366920938463463374607431768211456 does not fit into type uint128."
+    );
+    assert_eq!(errors[8].message, "exponent '256' too large");
```

Comment:
This does not test your implementation. The error comes from another place in sema. You could try instead:
```
uint256 num = 115792089237316195423570985008687907853269984665640564039457584007913129639935 * 4;
```